src/lib/stitching/vortex/schema.ts

```diff
@@ -20,41 +20,45 @@ export const executableVortexSchema = ({
     link: vortexLink,
   })
 
-  const removeRootFieldList = [
-    "pricingContext",
+  const rootFieldsToFilter = [
+    "BigInt",
+    "artistAffinities",
+    "artistRecommendations",
     "partnerStat",
+    "pricingContext",
     "userStat",
-    "BigInt",
   ]
 
-  // Return the new modified schema
-  return transformSchema(schema, [
-    // we don't want pricingContext to be a root query field, it is
-    // accessible through artwork
-    ...(removeRootFields
-      ? [
-          new FilterRootFields(
-            (_operation, name) => !removeRootFieldList.includes(name)
-          ),
-        ]
-      : []),
-    new RenameTypes((name) => {
-      if (
-        name.includes("PriceInsight") ||
-        name.includes("PageCursor") ||
-        ["BigInt", "ISO8601DateTime"].includes(name)
-      ) {
-        return name
-      } else {
-        return `Analytics${name}`
-      }
-    }),
-    new RenameRootFields((_operation, name) => {
-      if (["priceInsights", "marketPriceInsights"].includes(name)) {
-        return name
-      } else {
-        return `analytics${name.charAt(0).toUpperCase() + name.slice(1)}`
-      }
-    }),
-  ])
+  const filterTransform = new FilterRootFields(
+    (_operation, name) => !rootFieldsToFilter.includes(name)
+  )
+
+  const typesToNotRename = ["BigInt", "ISO8601DateTime"]
+  const patternToNotRename = new RegExp(/(PageCursor|PriceInsight)/)
+
+  const renameTypesTransform = new RenameTypes((name) => {
+    if (name.match(patternToNotRename) || typesToNotRename.includes(name)) {
+      return name
+    } else {
+      return `Analytics${name}`
+    }
+  })
+
+  const rootFieldsToNotRename = ["marketPriceInsights", "priceInsights"]
+
+  const renameRootFieldTransform = new RenameRootFields((_operation, name) => {
+    if (rootFieldsToNotRename.includes(name)) {
+      return name
+    } else {
+      return `analytics${name.charAt(0).toUpperCase() + name.slice(1)}`
+    }
+  })
+
+  const transforms = [
+    ...(removeRootFields ? [filterTransform] : []),
+    renameTypesTransform,
+    renameRootFieldTransform,
+  ]
+
+  return transformSchema(schema, transforms)
```

Comment:
Appreciate the refactor! but I would suggest keeping this as it was:
- the left is the pattern we use all through our stitched services and the code all looks roughly the same everywhere, and also matches the docs
- from a more personal pref perspective the left is fundamentally simpler and more straightforward. One can read straight down and it says quite literally exactly whats going on in a linear fashion that requires very little thought. The right now has lots of new variables to parse, regexes, stuff being defined and then later executed out of order, requiring visual jumping back and forth to grok and so on. 

If this bit had more inherent complexity I'd say 👍 but for what it is, and for the sake of MP-wide consistency we should just focus on the auth token aspects for now. 

Of course, this is non-blocking! Just my 2c at a glance.

Reply:
I pondered this overnight and decided to take your advice - did it like so:

```
$ git checkout main -- src/lib/stitching/vortex/schema.ts
```

Then I adjusted the `removeRootFieldList` to include `artistAffinities` and `artistRecommendations` to match the change introduced in my refactor.

I'd like to rebase this PR then so that git history will include the refactoring if we ever change our minds and want it back. Thanks @damassi!!